src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -348,6 +648,92 @@ private static double[] ComputeSequenceDeltas(IReadOnlyList<double> a, IReadOnly
+
+            if (model is TransformerChain<ITransformer> chain)
+            {
+                foreach (var transformer in chain.Reverse())
+                {
+                    if (transformer is ISingleFeaturePredictionTransformer singlePredictionTransformer)
+                    {
+                        lastTransformer = singlePredictionTransformer;
+                        break;
+                    }
+                }
+            }
+            else lastTransformer = model as ISingleFeaturePredictionTransformer;
+
+            env.CheckValue(lastTransformer, nameof(lastTransformer), "The model provided does not have a compatible predictor");
+
+            string featureColumnName = lastTransformer.FeatureColumnName;
+            var predictionTransformerGenericType = GetImplementedIPredictionTransformer(lastTransformer.GetType());
+
+            Type[] types = { predictionTransformerGenericType.GenericTypeArguments[0], typeof(TMetric), typeof(TResult) };
+            Type pfiGenericType = typeof(PermutationFeatureImportance<,,>).MakeGenericType(types);
+
+            object[] param = { env,
+                lastTransformer,
+                data,
+                resultInitializer,
+                evaluationFunc,
+                deltaFunc,
+                featureColumnName,
+                permutationCount,
+                useFeatureWeightFilter,
+                numberOfExamplesToUse
+            };
+
+            MethodInfo mi = pfiGenericType.GetMethod("GetImportanceMetricsMatrix", BindingFlags.Static | BindingFlags.Public);
+            var permutationFeatureImportance = (ImmutableArray<TResult>)mi.Invoke(null, param);
+
+            VBuffer<ReadOnlyMemory<char>> nameBuffer = default;
+            data.Schema[featureColumnName].Annotations.GetValue("SlotNames", ref nameBuffer);
+            var featureColumnNames = nameBuffer.DenseValues().ToList();
+
+            var output = new Dictionary<string, TResult>();
+            for (int i = 0; i < permutationFeatureImportance.Length; i++)
+            {
+                var name = featureColumnNames[i].ToString();
+
+                // If the slot wasn't given a name, default to just the slot number.
+                if (string.IsNullOrEmpty(name))
+                {
+                    name = $"Slot {i}";
+                }
+                output.Add(name, permutationFeatureImportance[i]);
+            }
+
+            return output.ToImmutableDictionary();
+        }
+
+        private static Type GetImplementedIPredictionTransformer(Type type)
+        {
+            foreach (Type iType in type.GetInterfaces())
+            {
+                if (iType.IsGenericType && iType.GetGenericTypeDefinition() == typeof(IPredictionTransformer<>))
+                {
+                    return iType;
+                }
+            }
+
+            throw new ArgumentException($"Type IPredictionTransformer not implemented by provided type", nameof(type));
```

Comment:
(nit) It would be good to put the type name in the exception message.

Reply:
Good idea. Its been added.